src/script/interpreter.cpp

```diff
@@ -1123,9 +1123,15 @@ class CTransactionSignatureSerializer {
                 // * if Tx has Sc support version and nIn belongs to the CSW inputs - only the CSW input being signed is serialized
                 // * otherwise skip CSW inputs
                 // Otherwise - serialize all CSW inputs
+#if 0
                 unsigned int nCswInputs = fAnyoneCanPay ?
                             (nIn < txTo.GetVin().size() ? 0 : 1) :
                             txTo.GetVcswCcIn().size();
+#else
+            // TODO - check this: support for SIGHASH_ANYONECANPAY in csw has been removed because of
+            //        active cert data vector below
+            unsigned int nCswInputs = txTo.GetVcswCcIn().size();
+#endif
```

Comment:
this is fine.
The only point of improvement, that on RPC level we may prevent the combination of `SIGHASH_ANYONECANPAY` and CSWs notifying users with a proper message.

Reply:
Ok, I will return error from RPC cmd signrawtransaction() in that case